src/oc_erchef/apps/data_collector/src/data_collector_app.erl

```diff
@@ -0,0 +1,37 @@
+%% -*- erlang-indent-level: 4;indent-tabs-mode: nil; fill-column: 92 -*-
+%% ex: ts=4 sw=4 et
+%% @author Ryan Cragun <ryan@chef.io>
+%% @author John Keiser <jkeiser@chef.io.
+%%
+%% Copyright 2016 Chef Software, Inc. All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+
+-module(data_collector_app).
+
+-behaviour(application).
+
+%% API
+-export([start/2,
+         stop/1
+        ]).
+
+start(_StartType, _StartArgs) ->
+    error_logger:info_msg("Starting Data Collector app.~n"),
+    supervisor:start_link({local, data_collector_sup}, data_collector_sup, []).
```

Comment:
Minor point: I think that in our other applications we often expose a start_link/0 in the supervisor that calls supervisor:start_link